selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/stack/CreateWorkspaceFromCentosNodeStackTest.java

```diff
@@ -0,0 +1,119 @@
+import static org.eclipse.che.selenium.pageobject.dashboard.NewWorkspace.Stack.CENTOS_NODEJS;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import java.util.List;
+import org.eclipse.che.selenium.core.client.TestWorkspaceServiceClient;
+import org.eclipse.che.selenium.core.user.DefaultTestUser;
+import org.eclipse.che.selenium.pageobject.Consoles;
+import org.eclipse.che.selenium.pageobject.Ide;
+import org.eclipse.che.selenium.pageobject.ProjectExplorer;
+import org.eclipse.che.selenium.pageobject.dashboard.CreateWorkspaceHelper;
+import org.eclipse.che.selenium.pageobject.dashboard.Dashboard;
+import org.openqa.selenium.By;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/** @author Skoryk Serhii */
+public class CreateWorkspaceFromCentosNodeStackTest {
+
+  private static final String WORKSPACE_NAME = generate("workspace", 4);
+  private static final String ANGULAR_PATTERNFLY_STARTER = "angular-patternfly-starter";
+  private static final String NODEJS_HELLO_WORLD = "nodejs-hello-world";
+  private static final String WEB_NODEJS_SIMPLE = "web-nodejs-simple";
+
+  private List<String> projects =
+      ImmutableList.of(ANGULAR_PATTERNFLY_STARTER, NODEJS_HELLO_WORLD, WEB_NODEJS_SIMPLE);
+
+  @Inject private Ide ide;
+  @Inject private Consoles consoles;
+  @Inject private Dashboard dashboard;
+  @Inject private DefaultTestUser defaultTestUser;
+  @Inject private ProjectExplorer projectExplorer;
+  @Inject private CreateWorkspaceHelper createWorkspaceHelper;
+  @Inject private TestWorkspaceServiceClient workspaceServiceClient;
+
+  @BeforeClass
+  public void setUp() {
+    dashboard.open();
+  }
+
+  @AfterClass
+  public void tearDown() throws Exception {
+    workspaceServiceClient.delete(WORKSPACE_NAME, defaultTestUser.getName());
+  }
+
+  @Test
+  public void checkWorkspaceCreationFromCentosNodeStack() {
+    createWorkspaceHelper.createWorkspaceFromStackWithProjects(
+        CENTOS_NODEJS, WORKSPACE_NAME, projects);
+
+    ide.switchToIdeAndWaitWorkspaceIsReadyToUse();
+
+    projectExplorer.waitProjectInitialization(ANGULAR_PATTERNFLY_STARTER);
+    projectExplorer.waitProjectInitialization(NODEJS_HELLO_WORLD);
+    projectExplorer.waitProjectInitialization(WEB_NODEJS_SIMPLE);
+  }
+
+  @Test(priority = 1)
+  public void checkAngularPatternfyStarterProjectCommands() {
+    consoles.executeCommandFromProjectExplorer(
+        ANGULAR_PATTERNFLY_STARTER,
+        BUILD_GOAL,
+        INSTALL_DEPENDENCIES_COMMAND_ITEM.getItem(ANGULAR_PATTERNFLY_STARTER),
+        "bower_components/font-awesome");
+    consoles.executeCommandFromProjectExplorer(
+        ANGULAR_PATTERNFLY_STARTER,
+        RUN_GOAL,
+        RUN_COMMAND_ITEM.getItem(ANGULAR_PATTERNFLY_STARTER),
+        "Waiting...");
+    consoles.checkWebElementVisibilityAtPreviewPage(By.xpath("//*[@id='pf-app']"));
```

Comment:
It's not quite clear what "//*[@id='pf-app']" is about. From this point it would be better to name the string somehow.